src/XMakeBuildEngine/Definition/ToolsetReader.cs

```diff
@@ -471,6 +486,113 @@ bool accumulateProperties
         }
 
         /// <summary>
+        /// Create a dictionary with standard properties.
```

Comment:
Does Mono support System.Configuration?  Seems like it would be a lot more data-driven (and less code) to just add toolsets via the .exe.config rather than having to hard-code all of this if so.